SingularityUI/app/actions/api/base.es6

```diff
@@ -46,9 +46,20 @@ export function buildApiAction(actionName, opts={}) {
 
       let options = optsFunc(...args);
       return fetch(config.apiRoot + options.url, _.extend({credentials: 'include'}, _.omit(options, 'url')))
-        .then(response => response.json())
-        .then(json => {
-          dispatch(success(json));
+        .then(response => {
+          if (response.status >= 200 && response.status < 300) {
+            if (response.headers.get('Content-Type') === 'application/json') {
+              return response.json().then(json => dispatch(success(json)));
+            } else {
+              return response.text().then(body => dispatch(success({response: body})));
+            }
+          } else {
+            if (response.headers.get('Content-Type') === 'application/json') {
+              return response.json().then(body => dispatch(error(body)));
+            } else {
+              return response.text().then(body => dispatch(error({message: body})));
+            }
+          }
```

Comment:
I think that's what we were trying initially - but the error returned ends up being some variant of `invalid json`, and there's no way to get to the original data.